File: python/ray.py
class Ray(object):
	def __init__(self, origin, direction, time):
		self.origin = origin
		self.direction = direction
		self.time = time

	def __repr__(self):
		return "origin: {self.origin} direction: {self.direction} time: {self.time}".format(self=self)

	def point_at_t(self, t):
		return self.origin + (self.direction * t)

File: rust/src/ray.rs
use vec3::Vec3;

#[derive(Debug)]
pub struct Ray {
  pub origin: Vec3,
  pub direction: Vec3,
  pub time: f64,
}

impl Ray {
  pub fn new(origin: Vec3, direction: Vec3, time: f64) -> Ray {
    Ray {
      origin,
      direction,
      time
    }
  }

  pub fn point_at_parameter(&self, t: f64) -> Vec3 {
    self.origin + self.direction * t
  }
}
